File: Data Structures & Algorithms/maximum-subarray/submission-1.cs
public class Solution {
    public int MaxSubArray(int[] nums) {

        int n = nums.Length;
        int[] dp = (int[])nums.Clone();
        for ( int i = 1 ; i < n; i++)
        {
            dp[i] = Math.Max(nums[i], nums[i]+ dp[i-1]);
        }
        int res = nums[0];
        
        foreach (int sum in dp)
        {
            res = Math.Max(res,sum);
        }

        return res; 

    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-1.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        res = []
        val = float("-infinity")
        for i in range(len(nums)-k+1):
            val = nums[i]
            for j in range(i,i+k):
                val = max(nums[j],val)
            res.append(val)
        return res
